File: Semester 5/Data Managment/Zi2Lab7/Lab7/src/Entity/Nauczyciel.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

/**
 * @ORM\Entity
 */
class Nauczyciel
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     */
    private $id;
    /**
     * @ORM\Column(type="string", length=255)
     */
    private $imie;
    /**
     * @ORM\Column(type="string", length=255)
     */
    private $nazwisko;
    /**
     * @ORM\Column(type="string", length=255)
     */
    private $tytul;
    /**
     * @ORM\OneToMany(targetEntity="Przedmiot", mappedBy="nauczyciel", cascade={"persist", "remove"})
     */
    private $przedmioty;

    /**
     * @ORM\OneToOne(targetEntity="Klasa", cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="klasa_id", referencedColumnName="id")
     */
    private $klasa;

    public function __construct(string $imie = null, string $nazwisko = null, string $tytul = null)
    {
        $this->imie = $imie;
        $this->nazwisko = $nazwisko;
        $this->tytul = $tytul;
        $this->przedmioty = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): void
    {
        $this->id = $id;
    }

    public function getImie(): ?string
    {
        return $this->imie;
    }

    public function setImie(string $imie): void
    {
        $this->imie = $imie;
    }

    public function getNazwisko(): ?string
    {
        return $this->nazwisko;
    }

    public function setNazwisko(string $nazwisko): void
    {
        $this->nazwisko = $nazwisko;
    }

    public function getTytul(): ?string
    {
        return $this->tytul;
    }

    public function setTytul(string $tytul): void
    {
        $this->tytul = $tytul;
    }

    /**
     * @return Collection|Przedmiot[]
     */
    public function getPrzedmioty(): Collection
    {
        return $this->przedmioty;
    }

    public function addPrzedmiot(Przedmiot $przedmiot): void
    {
        if (!$this->przedmioty->contains($przedmiot)) {
            $this->przedmioty[] = $przedmiot;
            $przedmiot->setNauczyciel($this);
        }
    }

    public function removePrzedmiot(Przedmiot $przedmiot): void
    {
        if ($this->przedmioty->removeElement($przedmiot)) {
            if ($przedmiot->getNauczyciel() === $this) {
                $przedmiot->setNauczyciel(null);
            }
        }
    }

    public function getKlasa(): ?Klasa
    {
        return $this->klasa;
    }

    public function setKlasa(Klasa $klasa): void
    {
        $this->klasa = $klasa;
    }
}
?>

File: Semester 5/Data Managment/Lab6/src/main/java/org/example/Nauczyciel.java
package org.example;
import jakarta.persistence.*;
import java.util.List;
import java.io.Serializable;

@Entity
public class Nauczyciel implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String imie;
    private String nazwisko;
    private String tytul;
    @OneToMany(cascade = CascadeType.ALL)
    private List<Przedmiot> przedmioty;
    @OneToOne(cascade = CascadeType.ALL)
    private Klasa klasa;

    public Nauczyciel() {
    }
    public Nauczyciel(String imie, String nazwisko, String tytul) {
        this.imie = imie;
        this.nazwisko = nazwisko;
        this.tytul = tytul;
    }

    public List<Przedmiot> getPrzedmioty() {return przedmioty;}
    public void setPrzedmioty(List<Przedmiot> przedmioty) {this.przedmioty = przedmioty;}

    public Klasa getKlasa() {return klasa;}
    public void setKlasa(Klasa klasa) {this.klasa = klasa;}

    public Long getId() {return id;}
    public void setId(Long id) {this.id = id;}

    public String getImie() {return imie;}
    public void setImie(String imie) {this.imie = imie;}

    public String getNazwisko() {return nazwisko;}
    public void setNazwisko(String nazwisko) {this.nazwisko = nazwisko;}

    public String  getTytul() {return tytul;}
    public void setTytul(String tytul) {this.tytul = tytul;}
}
